packages/sdk/src/dom.ts

```diff
@@ -206,18 +206,40 @@ export function loadScript(
   },
 ) {
   const { attrs = {}, createScriptHook } = info;
-  return new Promise<void>((resolve, _reject) => {
-    const { script, needAttach } = createScript({
-      url,
-      cb: resolve,
-      attrs: {
+
+  const createScriptWithAttrs = (attrs: Record<string, any>) => {
+    return new Promise<void>((resolve, reject) => {
+      const { script, needAttach } = createScript({
+        url,
+        cb: resolve,
+        attrs,
+        createScriptHook,
+        needDeleteScript: true,
+      });
+      needAttach && document.head.appendChild(script);
+      script.onerror = (event) => {
+        script.parentNode?.removeChild(script);
+        reject(event);
+      };
+    });
+  };
+
+  return createScriptWithAttrs(
+    Object.assign(
+      {
         crossorigin: 'anonymous',
         fetchpriority: 'high',
-        ...attrs,
       },
-      createScriptHook,
-      needDeleteScript: true,
-    });
-    needAttach && document.head.appendChild(script);
-  });
+      attrs,
+    ),
+  ).catch(() =>
+    createScriptWithAttrs(
```

Comment:
should we only re inject script for cors error  ? For other errors, we should throw errors as normal

Reply:
Or maybe we can remove `crossorigin: 'anonymous', ` attr, and add options to support config it

Reply:
yeah i was thinking we remove the crossorigin, since i cannot see if script error is 404 or CORS error from onerror (that i know of)  I think is we passed a `crossorigin` option for user to control, or enable it when one uses json manifest, its more acceptable